src/create-theme-provider.js

```diff
@@ -0,0 +1,92 @@
+const React = require('react');
+const PropTypes = require('prop-types');
+const isFunction = require('is-function');
```

Comment:
lets keep it, its implementation detail. we can discuss it in separate issue later